examples/console/main.go

```diff
@@ -72,88 +95,131 @@ func run() error {
+			err = createAccountWithRoot(horizonClient, networkDetails.NetworkPassphrase, accountKey)
+			if err != nil {
+				return err
+			}
+			account, err = horizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: accountKey.Address()})
+		}
+		if err != nil {
+			return err
+		}
+		accountSeqNum, err := account.GetSequenceNumber()
+		if err != nil {
+			return err
+		}
+
+		escrowAccountKeyFull := keypair.MustRandom()
+		escrowAccountKey = escrowAccountKeyFull.FromAddress()
+		fmt.Fprintln(os.Stdout, "escrow account:", escrowAccountKey.Address())
+
+		config := agentpkg.Config{
+			ObservationPeriodTime:      observationPeriodTime,
+			ObservationPeriodLedgerGap: observationPeriodLedgerGap,
+			MaxOpenExpiry:              maxOpenExpiry,
+			NetworkPassphrase:          networkDetails.NetworkPassphrase,
+			SequenceNumberCollector:    horizon,
+			BalanceCollector:           horizon,
+			Submitter:                  submitter,
+			Streamer:                   horizon,
+			EscrowAccountKey:           escrowAccountKey,
+			EscrowAccountSigner:        signerKey,
+			LogWriter:                  os.Stderr,
+			Events:                     events,
+		}
+		if filename != "" {
+			config.Snapshotter = JSONFileSnapshotter{
+				Filename:                   filename,
+				ObservationPeriodTime:      observationPeriodTime,
+				ObservationPeriodLedgerGap: observationPeriodLedgerGap,
+				MaxOpenExpiry:              maxOpenExpiry,
+				EscrowAccountKey:           escrowAccountKey,
+			}
+		}
+		agent = agentpkg.NewAgent(config)
+
+		tx, err := txbuild.CreateEscrow(txbuild.CreateEscrowParams{
+			Creator:        accountKey.FromAddress(),
+			Escrow:         escrowAccountKey.FromAddress(),
+			SequenceNumber: accountSeqNum + 1,
+			Asset:          txnbuild.NativeAsset{},
+		})
+		if err != nil {
+			return fmt.Errorf("creating escrow account tx: %w", err)
+		}
+		tx, err = tx.Sign(networkDetails.NetworkPassphrase, signerKey, escrowAccountKeyFull)
+		if err != nil {
+			return fmt.Errorf("signing tx to create escrow account: %w", err)
+		}
+		err = submitter.SubmitTx(tx)
+		if err != nil {
+			return fmt.Errorf("submitting tx to create escrow account: %w", err)
+		}
+	} else {
+		escrowAccountKey = file.EscrowAccountKey
+		config := agentpkg.Config{
+			ObservationPeriodTime:      file.ObservationPeriodTime,
+			ObservationPeriodLedgerGap: file.ObservationPeriodLedgerGap,
+			MaxOpenExpiry:              file.MaxOpenExpiry,
+			NetworkPassphrase:          networkDetails.NetworkPassphrase,
+			SequenceNumberCollector:    horizon,
+			BalanceCollector:           horizon,
+			Submitter:                  submitter,
+			Streamer:                   horizon,
+			EscrowAccountKey:           escrowAccountKey,
+			EscrowAccountSigner:        signerKey,
+			LogWriter:                  os.Stderr,
+			Events:                     events,
+		}
+		if filename != "" {
```

Comment:
💭  feels weird to me that there could be a case where file exists but filename is empty. Which would leave the config with an empty Snapshotter field.

Thoughts on erroring if filename isn't given on the cli? Alternatively adding a default filename name at the top eg. "demo-snapshot.json"

Reply:
Thanks for catching this, the `filename != ""` condition is unnecessary. If there's a `file` there is always a `filename`.